gr-qtgui/lib/fft_shift.h

```diff
@@ -0,0 +1,67 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2012,2014-2015 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * GNU Radio is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3, or (at your option)
+ * any later version.
+ *
+ * GNU Radio is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNU Radio; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifndef INCLUDED_QTGUI_FFT_SHIFT_H
+#define INCLUDED_QTGUI_FFT_SHIFT_H
+
+#include <algorithm>
+#include <vector>
+
+namespace gr {
+  namespace qtgui {
+
+    class fft_shift {
+    public:
+      fft_shift(size_t fft_length)
+	: d_fftlen()
```

Comment:
If you want to default-initialize these, I would recommend doing it downstairs instead.

Reply:
I realized there  is no need to default initialize and then to set the correct values. It is now done in the constructor in one go (not calling `resize`).